engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/ReentrantLocking.scala

```diff
@@ -45,16 +45,6 @@ class ReentrantLocking(logger: TruffleLogger) extends Locking {
     }
   }
 
-  /** @inheritdoc */
```

Comment:
I am not completely sure why `Locking` and its `ReentrantLocking` implementation are separated into two files. Possibly some attempt to write clean code ([code against interfaces, not implementation](http://wiki.apidesign.org/wiki/Code_Against_Interfaces%2C_Not_Implementations), maybe?). However given we are not going to be using any other than the single implementation, it might be better to turn `Locking` into `final class` and move the implementation in there.